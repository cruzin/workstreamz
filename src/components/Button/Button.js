import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './Button.scss';

const Button = ({ children, ...props }) => {
  const { primary, secondary, className, handleClick } = props;

  const classNames = cx(
    'button',
    { 'button--primary': primary, 'button--secondary': secondary },
    className,
  );

  return (
    <button type='button' className={classNames} onClick={handleClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.string,
  handleClick: PropTypes.func,
};

export default Button;
