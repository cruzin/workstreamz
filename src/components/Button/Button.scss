.button {
  padding: 10px 15px;
  border: none;
  color: #fff;
  cursor: pointer;
  font-weight: 600;
  border-radius: 5px;
  outline: none;

  &--primary {
    background: #2fbbf9;
  }

  &--secondary {
    background: #8bc34a;
  }
}
