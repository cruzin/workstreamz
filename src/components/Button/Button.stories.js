import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';

storiesOf('Components/Button', module)
  .add('Primary', () => <Button primary>Primary Button</Button>)
  .add('Secondary', () => <Button secondary>Secondary Button</Button>)
  .add('Click', () => (
    // eslint-disable-next-line no-alert
    <Button primary handleClick={() => alert('clicked')}>
      Click Button
    </Button>
  ));
