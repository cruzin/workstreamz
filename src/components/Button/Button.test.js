/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Submit</Button>, div);
  });

  it('renders markup correctly', () => {
    const tree = renderer.create(<Button>Submit</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
