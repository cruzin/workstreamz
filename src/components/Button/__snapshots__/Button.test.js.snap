// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders markup correctly 1`] = `
<button
  className="button"
  type="button"
>
  Submit
</button>
`;

exports[`Path renders markup correctly 1`] = `
<button
  className="button"
  type="button"
>
  Submit
</button>
`;
